### PR TITLE
Additions to `fix-import`

### PR DESCRIPTION
**Pull request by fredrik-bakke**
**Changes:**
- Sort by namespace first (i.e. `foundation` before `foundation-core`)
- Add line break between namespaces
- Remove repeat imports
- Remove redundant submodule imports if entire namespace is imported
- Remove `foundation-core` imports if the corresponding `foundation` module is imported

**Example** result of `fix-imports` (on `abstract-polytopes`):

```agda
open import elementary-number-theory

open import foundation.cartesian-product-types
open import foundation.contractible-types
open import foundation.dependent-pair-types
open import foundation.disjunction
open import foundation.identity-types
open import foundation.propositional-truncations
open import foundation.propositions
open import foundation.sets
open import foundation.unit-type
open import foundation.universe-levels

open import order-theory.finitely-graded-posets
open import order-theory.posets

open import univalent-combinatorics
```

### CODE REVIEW

**Comment by fredrik-bakke** [comment]
I haven't tested this PR in much detail yet.

File: src/category-theory/initial-objects-precategories.lagda.md [hunk 1]
```diff
@@ -8,10 +8,12 @@ module category-theory.initial-objects-precategories where
...
 ```agda
 open import category-theory.precategories
-open import foundation-core.identity-types
+
 open import foundation.contractible-types
 open import foundation.dependent-pair-types
 open import foundation.universe-levels
+
+open import foundation-core.identity-types
 ```
```
Comment (fredrik-bakke) [inline src/category-theory/initial-objects-precategories.lagda.md:17]: The namespace `foundation` comes before `foundation-core` alphabetically, but the string `foundation.` comes after `foundation-core.` 😬

**Comment by fredrik-bakke** [comment]
This PR no longer places namespace imports (e.g. `open import foundation`) before all other non-public imports.

File: hooks/fix-imports.py [hunk 2]
```diff
@@ -2,48 +2,78 @@
...
+            namespaces[__import[:__import.index(".")]].add(__import)
+        except ValueError:
+            namespaces[__import].add(__import)
+
+    # If the entire namespace is imported, no further imports from that namespace are needed
+    for k in namespaces.keys():
+        if k in namespaces[k]:
+            namespaces[k] = {k}
+
+    for __import in namespaces["open import foundation"]:
```
Comment (jonaprieto) [inline hooks/fix-imports.py:25]: - Use shorter keys for fast dict access; remove "open import". You can put in the `join` call below.

File: hooks/fix-imports.py [hunk 3]
```diff
@@ -2,48 +2,78 @@
...
         start = contents.find('<details>')
         end = contents.find('</details>')
         if start != -1 and end != -1:
             block = contents[start:end]
-            newBlock = filter(lambda l: l.strip() != '', block.split('\n'))
-            publicImports = []
-            nonPublicImports = []
-            otherStuff = []
+            # Strip all lines. There should not be any indentation in this block
+            newBlock = filter(lambda l: l != '', map(lambda l: l.strip(), block.split('\n')))
```
Comment (jonaprieto) [inline hooks/fix-imports.py:42]: Use `.lstrip` and `.rstrip` here instead of below.

Comment (fredrik-bakke) [reply hooks/fix-imports.py:48]: Thanks, I was undecided about where to do it when I wrote the code which is why it was overdone.

**Review by jonaprieto — APPROVED** [review]

**Pull request merged**